modules/platforms/cpp/core-test/src/compute_security_test.cpp

```diff
@@ -0,0 +1,187 @@
+                                                                                                        \
+    BOOST_ASSERT(errCode != -1);                                                                        \
+}                                                                                                       \
+
+/** */
+struct ComputeSecurityTestSuiteFixture {
+    /** */
+    Ignite node;
+
+    /** */
+    ComputeSecurityTestSuiteFixture() : node(StartNode("compute-security.xml", "test-node")) {
+        // No-op.
+    }
+
+    /** */
+    ~ComputeSecurityTestSuiteFixture()
+    {
+        Ignition::StopAll(true);
+    }
+};
+
+struct AbstractCallable : ComputeFunc<int> {
+    virtual int Call() {
+        return 42;
+    }
+};
+
+/** */
+struct AllowedCallable : AbstractCallable { };
+
+/** */
+struct ForbiddenCallable : AbstractCallable { };
+
+/** */
+struct AbstractRunnable : ComputeFunc<void> {
+    virtual void Call() {
+        // No-op.
+    }
+};
+
+/** */
+struct AllowedRunnable : AbstractRunnable { };
+
+/** */
+struct ForbiddenRunnable : AbstractRunnable { };
+
+namespace ignite {
+    namespace binary {
+        template<>
+        struct BinaryType<AllowedCallable> : BinaryTypeDefaultAll<AllowedCallable> {
+            static void GetTypeName(std::string &dst) {
+                dst = "AllowedCallable";
+            }
+
+            static void Write(BinaryWriter&, const AllowedCallable&) {}
+
+            static void Read(BinaryReader&, AllowedCallable&) {}
+        };
+
+        template<>
+        struct BinaryType<ForbiddenCallable> : BinaryTypeDefaultAll<ForbiddenCallable> {
+            static void GetTypeName(std::string &dst) {
+                dst = "ForbiddenCallable";
+            }
+
+            static void Write(BinaryWriter&, const ForbiddenCallable&) {}
+
+            static void Read(BinaryReader&, ForbiddenCallable&) {}
+        };
+
+        template<>
+        struct BinaryType<AllowedRunnable> : BinaryTypeDefaultAll<AllowedRunnable> {
+            static void GetTypeName(std::string &dst) {
+                dst = "AllowedRunnable";
+            }
+
+            static void Write(BinaryWriter&, const AllowedRunnable&) {}
+
+            static void Read(BinaryReader&, AllowedRunnable&) {}
+        };
+
+        template<>
+        struct BinaryType<ForbiddenRunnable> : BinaryTypeDefaultAll<ForbiddenRunnable> {
+            static void GetTypeName(std::string &dst) {
+                dst = "ForbiddenRunnable";
+            }
+
+            static void Write(BinaryWriter&, const ForbiddenRunnable&) {}
+
+            static void Read(BinaryReader&, ForbiddenRunnable&) {}
+        };
+    }
+}
+
+BOOST_FIXTURE_TEST_SUITE(ComputeSecurityTestSuite, ComputeSecurityTestSuiteFixture)
+
+    BOOST_AUTO_TEST_CASE(TestComputeSecurity) {
+        Compute compute = node.GetCompute();
+
+
+        BOOST_ASSERT(42 == compute.Call<int>(AllowedCallable()));
```

Comment:
Here and in other places:
```suggestion
        BOOST_CHECK_EQUAL(42, compute.Call<int>(AllowedCallable()));
```

Reply:
Fixed.